src/components/dashboard/amend-balance.svelte

```diff
@@ -0,0 +1,62 @@
+<script>
+  import { getContext } from 'svelte';
+  import { TextField, Button } from 'attractions';
+  import { snackbarContextKey } from 'attractions/src/snackbar';
+  import * as api from '@/utils/api.js';
+  import s from '@/utils/plural-s.js';
+
+  const showSnackbar = getContext(snackbarContextKey);
+
+  export let users = [];
+  export let account;
+
+  let amount = '';
+
+  async function send() {
+    try {
+      const promises = users.map(user => api.json(api.patch(`/accounts/${user.email}/balance`, {
+        data: {
+          change: amount,
+        },
+        csrfToken: account.csrf_token,
+      })));
+      await Promise.all(promises);
+      showSnackbar({
```

Comment:
Wouldn't it be better if a snackbar is shown for each email so that if some succeed and others fail, it would be easy to know what happened to each and not send duplicate points?
`Promise.allSettled` can help here, but it is relatively new. Maybe attaching `.then` and `.catch` to each promise directly would be better

Reply:
It would get really messy. Good point, but z
I think it's infeasible in practice. Besides, it shouldn't occur often that one fails and the other succeeds

Reply:
I wouldn't say it is infeasible (it is very easy to implement). Perhaps it can get messy, especially when all succeed or all fail, which are the most common cases (I agree). We can check for that and collect successes/failures in one snackbar.
Alright, it's not worth it.